src/config/resolver.rs

```diff
@@ -0,0 +1,111 @@
+use std::collections::VecDeque;
+
+use anyhow::anyhow;
+use serde::{Deserialize, Serialize};
+use tokio::fs::File;
+use tokio::io::AsyncReadExt;
+use url::Url;
+
+use crate::config::{is_default, Config, Source};
+
+#[derive(Default, Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
+pub enum LinkType {
+  #[default]
+  Config,
+  GraphQL,
+  Protobuf,
+  Data,
+}
+
+#[derive(Default, Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
+pub struct Link {
+  #[serde(default, skip_serializing_if = "is_default", rename = "type")]
+  pub type_of: LinkType, // Type of the link
+  #[serde(default, skip_serializing_if = "is_default")]
+  pub src: String, // Source URL for linked files
+  #[serde(default, skip_serializing_if = "is_default")]
+  pub id: Option<String>, // Id is used to refer at different places in the config
+  content: Option<String>, // Stores raw content
+}
+
+impl Link {
+  fn merge_configs(config: Option<Config>, config_right: Option<Config>) -> anyhow::Result<Option<Config>> {
+    match (config, config_right) {
+      (Some(c), Some(cc)) => Ok(Some(c.merge_right(&cc)?)),
+      (Some(c), None) => Ok(Some(c)),
+      (None, Some(cc)) => Ok(Some(cc)),
+      (None, None) => Ok(None),
+    }
+  }
+
+  pub async fn resolve_recurse(config_links: &mut Vec<Link>) -> anyhow::Result<Option<Config>> {
```

Comment:
```suggestion
  pub async fn resolve(config: &Config) -> anyhow::Result<Config> {
```

the `resolve` resolve the provided config and replace the `links` with `link_contents`.